tests/system/utils.py

```diff
@@ -39,3 +40,122 @@ def get_resource(resource):
                 raise Exception
         except Exception:
             raise DCOSException("Can't read from resource: {0}. Please check that it exists.".format(resource))
+
+
+class FaultDomain:
+    """
+    High-level representation of a fault domain
+    """
+    def __init__(self, config):
+        # Make sure config is a dict
+        if not type(config) is dict:
+            config = {}
+
+        # Extract fault domain
+        fault_domain = config.get('fault_domain', {})
+
+        # Extract zone/region
+        self.zone = fault_domain.get('zone', {}).get('name', 'default')
+        self.region = fault_domain.get('region', {}).get('name', 'default')
+
+
+def get_cluster_local_domain():
+    """Contacts the DC/OS mesos master and returns it's faultDomain configuration (aka "local domain")
+    """
+    master = mesos.get_master()
+    return FaultDomain(master.state().get('domain', {}))
+
+
+def get_cluster_agent_domains():
+    """Returns a dictionary with the slave IDs in the cluster and their corresponding
+       fault domain information
+    """
+    agent_domains = {}
+
+    # Populate agent_domains with the ID and the corresponding domain for each slave
+    master = mesos.get_master()
+    for slave in master.slaves():
+        agent_domains[slave['id']] = FaultDomain(slave._short_state.get('domain', None))
+    return agent_domains
+
+
+def get_all_cluster_regions():
+    """Returns a dictionary with all the regions and their zones in the cluster
+    """
+    domain_regions = {}
+
+    # Populate all the domain zones and regions found in the cluster
+    for domain in get_cluster_agent_domains().values():
+        if domain.region not in domain_regions:
+            domain_regions[domain.region] = []
+        if domain.zone not in domain_regions[domain.region]:
+            domain_regions[domain.region].append(domain.zone)
+
+    # Return dictionary
+    return domain_regions
+
+
+def get_biggest_cluster_region():
+    """Returns a tuple with the name of the region with the most zones in the
+       cluster and a list with the actual zones in it
+    """
+    biggest_region = None
+    biggest_region_zones = []
+
+    for (region, zones) in get_all_cluster_regions().items():
+        if len(zones) > len(biggest_region_zones):
+            biggest_region_zones = zones
+            biggest_region = region
```

Comment:
I'm not sure if we really need `get_biggest_cluster_region` but what do you think about
```python
def get_biggest_cluster_region():
   return max(get_all_cluster_regions().items(), key=lambda kv: len(kv[1]))
```